### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by amashutikov**
- [DEMO LINK](https://amashutikov.github.io/layout_catalog_grid/)

### CODE REVIEW

File: src/styles/blocks/_catalog.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+.catalog {
+  display: grid;
+  gap: 48px;
+  grid-template-columns: $card-width $card-width $card-width $card-width;
```
Comment (ghost) [inline src/styles/blocks/_catalog.scss]: what if we need more columns? let's use `repeat()` function

File: src/styles/blocks/_catalog.scss [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  @media (max-width: 767px) {
+    grid-template-columns: $card-width $card-width;
+  }
+
+  @media (max-width: 487px) {
+    grid-template-columns: $card-width;
+  }
+}
```
Comment (ghost) [inline src/styles/blocks/_catalog.scss]: it would be better to make mobile grids first and go to the highest resolution

Comment (ghost) [reply src/styles/blocks/_catalog.scss]: also you can try to implement this functionality with a single loop

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by naziamoff — APPROVED** [review]
🔥

**Pull request closed without merging**